feat: [IDP-3610]: Docs for homepage customisation (#7786)
harness/developer-hub: docs/internal-developer-portal/layout-and-appearance/home-page-customization.md

---
title: Customize your Homepage
sidebar_label: Homepage Customization
description: Learn how you can customize your homepage and add card banners and headers of your choice
sidebar_position: 1
---

## Introduction

> Homepage of IDP is meant to provide personalized view for developers, serving as a single pane of glass for all the necessary information required across tools and software systems in use and it differs for different organization as well as different teams in an organization. So to help with varied use-cases we now enable customizing the homepage, adding headers, banners and cards according to your engineering org making it useful for your developers. 

Reviewer: ```suggestion
Homepage of IDP is meant to provide personalized view for developers, serving as a single pane of glass for all the necessary information required across tools and software systems in use and it differs for different organization as well as different teams in an organization. So to help with varied use-cases, you can customize the homepage, add quick links in the headers, change text on the header, banner and cards according to your engineering org and make it useful for your developers. 
```

## Available Customizations

The **Platform Admins** can customize the homepage under Layouts for 3 different categories **Header**, **Banner** and **Cards**. A preview of the changes with sample data is available while customizing the homepage. The cards can be of two width sizes, i.e., medium(6md) and large(12md).

### Header

- Under Header you could add the personalized welcome message including quick links to important pages like Release Notes, Engineering Docs and Status Page for important service etc. 

![](./static/homepage-header.png)

> - For Header Text, enter the values in the field given, which could be dynamic as well like `<+first_name> <+last_name>`. 

Reviewer: Worth adding all the possibilities here.

1. You can leave it static such as - Welcome to Harness IDP!
2. You can make it personalized such as - Welcome <+first_name>!
3. If your users have names of the format "last name, first name", then you can use Welcome <+last_name>, <+first_name>!
4. Or you can name the portal a unique product name such as - Welcome to MyPortal! (maybe, a bit more creative than MyPortal).
5. There is a special variable called <+greetings> which resolves to Good Morning, Good Evening, etc. depending upon the time for the user. e.g. <+greetings> <+first_name>

> - For Quick links you can add the `custom icon` **(Recommended Size: 128x128px and volume: 200kb)** along with a name and a link. You could as well drag and re-arrange the Quick Links cards once added and can preview them. 

Reviewer: ```suggestion
- For Quick links you can upload a Custom Icon **(Recommended Size: 128x128px and file size: 200KB)** along with a name and a link. You can also drag and re-arrange the order of the links. It's ideal to add a few links to some central documents that you want your users to go to when they come to your Developer Portal. e.g. Onboarding Docs, Link to Slack, etc.
```

Reviewer: is there a max file size limit here for the icons?

![](./static/add-quicklinks.png)

### Banner

> - You can choose to show a banner by default you have the IDP image as a banner. Choose between Image or Video.

Reviewer: ```suggestion
- You can replace the existing banner on the IDP homepage by either an Image or a Video.
```

>     - Image: Supported Types - `jpg, .jpeg, .png` ; Supported Volume - `10mb`; Recommended Size - `756x300px`  You can add an hyperlink to the image as well.

Reviewer: ```suggestion
    - Image: Supported Types - `.jpg`, `.jpeg` and `.png`. Recommended Size of the image is 756x300px Maximum file size allowed is 10MB. You can also add a hyperlink to the image as well which will take users to the link when they click the banner. Suggestion - Add a cool banner for an upcoming engineering event that you want developers to be aware of!
```

>     - Videos: Supported Types - Public Embedded Links eg; `https://www.youtube.com/embed/sVnI93bCr38?si=zobQ1YJMVVJMccaO`, `https://www.dropbox.com/scl/fo/5pu6lcznlqushows1gluk/AJtpk6i0ze_7Zr3Xr4xc2DE/distributed-work-with-dropbox.jpg?rlkey=d113ffnzhu8vseecxxe61ddk3&e=1&st=xzuy6mcy&raw=1` ; Rendered Size - `500x250px`. 

Reviewer: does it have to be public? what happens if it's an embed link from google drive which i have access to?

![](./static/homepage-banner.png)

### Cards

![](./static/homepage-cards.png)

#### Default Cards

- The home page displays the following cards by default and they can only be removed or rearranged by using drag-drop but can't be edited.

> 1. Recently Visited: The pages on IDP you visited recently.

Reviewer: ```suggestion
1. Recently Visited: The entities and pages in IDP that you visited recently.
```

> 2. Top Visited: Most viewed pages in IDP by an user.

Reviewer: ```suggestion
2. Top Visited: Most viewed entities or pages in IDP by you.
```

3. Learn More: Important links to learn more on Harness IDP. 

> 4. Starred Entities: On the catalog page you have an option to Star your favorite entities and these are displayed here. 

Reviewer: ```suggestion
4. Starred Entities: Your Starred Components, Workflows or TechDocs are displayed here. 
```

#### Other Cards

> Apart from these, there are two more cards that can be added, removed and edited according to the customized requirements. 

Reviewer: ```suggestion
Apart from these, there are two more cards that can be added, removed or edited according to your requirements. 
```

### Tool Kit

-  This card acts as a necessary bookmark for all the links visited by the developer quite frequently, eg; JIRA Sprint Dashboard, QA Environments

-  You can add a new toolkit card with multiple links and the associated `custom icon` **(Recommended Size: 128x128px and volume: 200kb)** along with a name and a link. 

### Video

> - You can add any videos as a card, and the supported types are public embedded links of the videos. 

Reviewer: Same as above, can this be an internal drive link of a video?
